bundles/org.palladiosimulator.editors.sirius.ui/src/org/palladiosimulator/editors/sirius/ui/wizard/project/NewPalladioProjectWizard.java

```diff
@@ -214,49 +218,80 @@ private void createAndOpenProject(final IProjectDescription description, final I
+        private void addFolderToProject(final URI path, final File source, final IFolder target,
+                final SubMonitor subMonitor) throws CoreException {
+            if (!target.exists()) {
+                target.create(IResource.NONE, true, null);
+            }
+
+            addToProject(path.appendSegment(source.getName()), target, subMonitor);
         }
 
-        addToProject(path.appendSegment(source.getName()), target, subMonitor);
-    }
+        private void addFileToProject(final File source, final IFile target, final SubMonitor subMonitor)
+                throws CoreException {
+            try (final InputStream contentStream = loadInputFile(source)) {
+                if (target.exists()) {
+                    target.setContents(contentStream, true, true, subMonitor);
+                } else {
+                    target.create(contentStream, true, subMonitor);
+                }
+            } catch (final FileNotFoundException e) {
+                throwCoreException("File " + source.getAbsolutePath() + " does not exist!");
+            } catch (final IOException e) {
+                throwCoreException(
+                        "Cannot create input stream on file " + source.getAbsolutePath() + "! " + e.getMessage());
+            } catch (final CoreException e) {
+                throwCoreException(e.getMessage());
+            }
+        }
 
-    private void addFileToProject(final File source, final IFile target, final SubMonitor subMonitor)
-            throws CoreException {
-        try (final InputStream contentStream = new FileInputStream(source)) {
-            if (target.exists()) {
-                target.setContents(contentStream, true, true, subMonitor);
-            } else {
-                target.create(contentStream, true, subMonitor);
+        private InputStream loadInputFile(File source) throws IOException {
+            // launch configurations should be small enough to load
+            if (source.getName()
+                .endsWith(".launch")) {
+                String fileContent = Files.readString(source.toPath());
+                fileContent = fileContent.replace(createPlatformUriStart(template.getEntityName()),
+                        createPlatformUriStart(projectHandle.getName()));
+                return new ByteArrayInputStream(fileContent.getBytes());
             }
-        } catch (final FileNotFoundException e) {
-            throwCoreException("File " + source.getAbsolutePath() + " does not exist!");
-        } catch (final IOException e) {
-            throwCoreException(
-                    "Cannot create inpht stream on file " + source.getAbsolutePath() + "! " + e.getMessage());
-        } catch (final CoreException e) {
-            throwCoreException(e.getMessage());
+
+            return new FileInputStream(source);
         }
     }
 
+    private static String createPlatformUriStart(String projectName) {
+        return "platform:/resource/" + projectName + "/";
```

Comment:
You might want to use the URI class to create a platform resource URI and convert it to a string afterwards. That would avoid the magic string.